sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/TopicSubscriptionWithRuleOperationsSample.java

```diff
@@ -0,0 +1,210 @@
+        administrationClient.updateRule(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_ONLY_SUBSCRIPTION_NAME,
+            administrationClient.getRule(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_ONLY_SUBSCRIPTION_NAME, SQL_FILTER_ONLY_SUBSCRIPTION_RULE_NAME)
+                .setFilter(new SqlRuleFilter("Color = 'Red'")));
+
+        // 3rd Subscription: Add SqlFilter and SqlRuleAction on Subscription 3
+        // Delete Default Rule
+        // Add the required SqlFilter Rule and Action
+        System.out.println(String.format("SubscriptionName: %s, Removing Default Rule and Adding CorrelationFilter", SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME));
+        administrationClient.deleteRule(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME);
+        administrationClient.createRule(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_RULE_NAME);
+        administrationClient.updateRule(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME,
+            administrationClient.getRule(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_RULE_NAME)
+                .setFilter(new SqlRuleFilter("Color = 'Blue'"))
+                .setAction(new SqlRuleAction("SET Color = 'BlueProcessed'")));
+
+        // 4th Subscription: Add Correlation Filter on Subscription 4
+        System.out.println(String.format("SubscriptionName: %s, Removing Default Rule and Adding CorrelationFilter", CORRELATION_FILTER_SUBSCRIPTION_NAME));
+        administrationClient.deleteRule(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME);
+        administrationClient.createRule(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME, CORRELATION_FILTER_SUBSCRIPTION_RULE_NAME);
+        administrationClient.updateRule(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME,
+            administrationClient.getRule(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME, CORRELATION_FILTER_SUBSCRIPTION_RULE_NAME)
+                .setFilter(new CorrelationRuleFilter().setCorrelationId("important").setLabel("Red")));
+
+        // Get Rules on Subscription, called here only for one subscription as example
+        administrationClient.listRules(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME)
+            .forEach(ruleProperties -> {
+                System.out.println(String.format("GetRules:: SubscriptionName: %s, CorrelationFilter Name: %s, Rule: %s", CORRELATION_FILTER_SUBSCRIPTION_NAME, ruleProperties.getName(), ruleProperties.getFilter()));
+            });
+
+        // Send messages to Topic
+        sendMessagesAsync();
+
+        // Receive messages from 'allMessagesSubscriptionName'. Should receive all 9 messages
+        receiveMessagesAsync(ALL_MESSAGES_SUBSCRIPTION_NAME);
```

Comment:
This is not an asynchronous function, I would remove the `async` keyword.

Reply:
According to your comment, fixed within new version